engine/classes/ElggAnnotation.php

```diff
@@ -75,12 +74,23 @@ public function save() {
 	 * @return bool
 	 */
 	public function delete() {
-		$result = _elgg_delete_metastring_based_object_by_id($this->id, 'annotation');
-		if ($result) {
+		if (!$this->canEdit()) {
```

Comment:
shouldn't this be `canDelete`

Reply:
I don't think we have canDelete on annotations